### PR TITLE
adding deviation for cisco delimiter, and skipping empty string 

### PR DESCRIPTION
**Pull request by fmolinar**
Adding the deviation for the banner test as Cisco devices expect a delimiter to be sent, and skipping the empty string banner test since Cisco currently doesnt support empty string send as banner.

### CODE REVIEW

File: feature/system/tests/system_base_test/banner_test.go [hunk 1]
```diff
@@ -30,31 +31,32 @@ import (
...
 			gnmi.Replace(t, dut, config.Config(), testCase.banner)
 
 			t.Run("Get MOTD Config", func(t *testing.T) {
-				if testCase.banner == "" {
-					if gnmi.LookupConfig(t, dut, config.Config()).IsPresent() {
-						t.Errorf("MOTD Banner not empty")
-					} else {
-						t.Logf("No response for the path is expected as the config is empty")
+
+				if testCase.description == "Empty String" {
```
Comment (robshakir) [inline feature/system/tests/system_base_test/banner_test.go:53]: We should not be matching on the description of a test case. This should be based on the content if we need to.

File: feature/system/tests/system_base_test/banner_test.go [hunk 2]
```diff
@@ -66,15 +68,18 @@ func TestMotdBanner(t *testing.T) {
 			})
 
 			t.Run("Get MOTD Telemetry", func(t *testing.T) {
-				if testCase.banner == "" {
-					if gnmi.LookupConfig(t, dut, config.Config()).IsPresent() {
-						t.Errorf("MOTD Telemetry Banner not empty")
-					} else {
-						t.Logf("No response for the path is expected as the config is empty")
+				if testCase.description == "Empty String" {
```
Comment (robshakir) [inline feature/system/tests/system_base_test/banner_test.go:71]: Ditto the above.

File: feature/system/tests/system_base_test/banner_test.go [hunk 3]
```diff
@@ -66,15 +68,18 @@ func TestMotdBanner(t *testing.T) {
...
+					if dut.Vendor() != ondatra.CISCO {
```
Comment (robshakir) [inline feature/system/tests/system_base_test/banner_test.go:72]: Please do not use vendor matches, use deviations to indicate non-compliance.

File: feature/system/tests/system_base_test/banner_test.go [hunk 4]
```diff
@@ -97,31 +102,30 @@ func TestMotdBanner(t *testing.T) {
...
 			gnmi.Replace(t, dut, config.Config(), testCase.banner)
 
 			t.Run("Get Login Banner Config", func(t *testing.T) {
-				if testCase.banner == "" {
-					if gnmi.LookupConfig(t, dut, config.Config()).IsPresent() {
-						t.Errorf("Config Login Banner not empty")
-					} else {
-						t.Logf("No response for the path expected is expected as the config is empty")
+				if testCase.description == "Empty String" {
+					if dut.Vendor() != ondatra.CISCO {
```
Comment (robshakir) [inline feature/system/tests/system_base_test/banner_test.go:123]: Same comments as above, do not use vendor matches or matches on test case descriptions. This should all be gated by deviations.

**Review by robshakir — CHANGES_REQUESTED** [review]

File: feature/system/tests/system_base_test/banner_test.go (hunk 1 above)
Comment (fmolinar) [reply feature/system/tests/system_base_test/banner_test.go]: I made changes in the new PR

File: feature/system/tests/system_base_test/banner_test.go (hunk 3 above)
Comment (fmolinar) [reply feature/system/tests/system_base_test/banner_test.go]: I created a deviation in the new PR

**Comment by mastarkey** [comment]
Hi @robshakir can you trigger the CI/CD checks here so we have an idea if those are passing.

**Comment by fmolinar** [comment]
outdated
